keras_nlp/src/models/efficientnet/efficientnet_backbone.py

```diff
@@ -0,0 +1,577 @@
+            name="stem_conv",
+        )(x)
+
+        x = keras.layers.BatchNormalization(
+            momentum=bn_momentum,
+            name="stem_bn",
+        )(x)
+        x = keras.layers.Activation(activation, name="stem_activation")(x)
+
+        # Build blocks
+        block_id = 0
+        blocks = float(sum(stackwise_num_repeats))
+
+        self._pyramid_outputs = {}
+        curr_pyramid_level = 1
+
+        for i in range(len(stackwise_kernel_sizes)):
+            num_repeats = stackwise_num_repeats[i]
+            input_filters = stackwise_input_filters[i]
+            output_filters = stackwise_output_filters[i]
+
+            # Update block input and output filters based on depth multiplier.
+            input_filters = round_filters(
+                filters=input_filters,
+                width_coefficient=width_coefficient,
+                min_depth=min_depth,
+                depth_divisor=depth_divisor,
+                use_depth_divisor_as_min_depth=use_depth_divisor_as_min_depth,
+                cap_round_filter_decrease=cap_round_filter_decrease,
+            )
+            output_filters = round_filters(
+                filters=output_filters,
+                width_coefficient=width_coefficient,
+                min_depth=min_depth,
+                depth_divisor=depth_divisor,
+                use_depth_divisor_as_min_depth=use_depth_divisor_as_min_depth,
+                cap_round_filter_decrease=cap_round_filter_decrease,
+            )
+
+            repeats = round_repeats(
+                repeats=num_repeats,
+                depth_coefficient=depth_coefficient,
+            )
+            strides = stackwise_strides[i]
+            squeeze_and_excite_ratio = stackwise_squeeze_and_excite_ratios[i]
+
+            for j in range(repeats):
+                # The first block needs to take care of stride and filter size
+                # increase.
+                if j > 0:
+                    strides = 1
+                    input_filters = output_filters
+
+                if strides != 1:
+                    self._pyramid_outputs[f"P{curr_pyramid_level}"] = x
+                    curr_pyramid_level += 1
+
+                # 97 is the start of the lowercase alphabet.
+                letter_identifier = chr(j + 97)
+                stackwise_conv_type = stackwise_conv_types[i]
+                if stackwise_conv_type == "v1":
+                    x = self._apply_efficientnet_block(
+                        inputs=x,
+                        filters_in=input_filters,
+                        filters_out=output_filters,
+                        kernel_size=stackwise_kernel_sizes[i],
+                        strides=strides,
+                        expand_ratio=stackwise_expansion_ratios[i],
+                        se_ratio=squeeze_and_excite_ratio,
+                        activation=activation,
+                        dropout_rate=drop_connect_rate * block_id / blocks,
```

Comment:
agreed, consolidated to "dropout" ... it seems the original implementation of those blocks had a bug ... they were using survival_probability as dropout -- but since the default was always set to 0.2 via this backbone it effectively never presented itself and acted appropriately.